src/NLog/LayoutRenderers/WindowsIdentityLayoutRenderer.cs

```diff
@@ -81,37 +81,26 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
             if (currentIdentity != null)
             {
                 string output = string.Empty;
-
-                if (UserName)
+                if (UserName && Domain)
+                {
+                    // username && domain
+                    output = currentIdentity.Name;
+                }
+                else if (UserName)
```

Comment:
I think this is a code smell (party same condition as previous condition)

Reply:
I just wanted to reduce the complexity. If you are welcome to continue the work in a new PR :)

Reply:
Yes I like that, but we need to be careful not introducing more fragile code :)

Reply:
Have you looked at the original code?

Reply:
Only diff (on mobile), but I will first then look in depth (and I could also commit on this Branch ;)

Reply:
I checked it in-depth, and will try to explain better:

The current code had the structure:

```C#
if(Username)
{
    if (Domain) { }
}
else
{
    if (!Domain) { }
}
```

The new code has the structure:

```C#
if (UserName && Domain) { }
else if (UserName) { }
else if (Domain) { }
```

I think the first structure is better because:
- easier conditions (no combinations)
- no else condition that also is already check in the if condition
- 2 small code blocks are preferred over 1 larger code block.


Still we need to refactor the code, as it was too long indeed.

I propose an extract method for this. See also https://blog.codinghorror.com/flattening-arrow-code/

Reply:
Well I like my code. Removes 10 lines of unnecessary code. And is more readable to me. Will revert my improvement.

Reply:
I have created: https://github.com/NLog/NLog/pull/3265

Reply:
Heheh so instead of removing 10 lines, then you have added 20 lines more. Ohh well.

Reply:
👼  the code complexity of all those methods is very low.

Reply:
Removed my changes to `WindowsIdentityLayoutRenderer.cs` from this PR